RetailCoder.VBE/Refactorings/ExtractMethod/ExtractMethodModel.cs

```diff
@@ -165,55 +129,62 @@ public string PreviewCode
         {
             get
             {
-                //var rewriter = State.GetRewriter(CodeModule.GetQualifiedSelection().Value.QualifiedName);
-
-                var fields = new List<string>();
-                var parameters = new List<string>();
-                var variables = new List<string>();
+                _fieldsList = new List<string>();
+                _parametersList = new List<string>();
+                _variablesList = new List<string>();
 
                 foreach (var parameter in Parameters)
                 {
                     switch (parameter.ParameterType)
                     {
-                        case ExtractParameterNewType.PublicModuleField:
-                            fields.Add(String.Format($"{Tokens.Public} {parameter.Name} {Tokens.As} {parameter.TypeName}"));
+                        case ExtractMethodParameterType.PublicModuleField:
+                            _fieldsList.Add(parameter.ToString(ExtractMethodParameterFormat.DimOrParameterDeclarationWithAccessibility));
                             break;
-                        case ExtractParameterNewType.PrivateModuleField:
-                            fields.Add(String.Format($"{Tokens.Private} {parameter.Name} {Tokens.As} {parameter.TypeName}"));
+                        case ExtractMethodParameterType.PrivateModuleField:
+                            _fieldsList.Add(parameter.ToString(ExtractMethodParameterFormat.DimOrParameterDeclarationWithAccessibility));
```

Comment:
consider collapsing case-blocks with the same content:

    case ExtractMethodParameterType.PublicModuleField:
    case ExtractMethodParameterType.PrivateModuleField:
        _fieldsList[....];
        break;